src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -96,13 +93,27 @@ static ClientApplicationBase()
         /// Gets or sets correlation Id which would be sent to the service with the next request.
         /// Correlation Id is to be used for diagnostics purposes.
         /// </summary>
-        public Guid CorrelationId { get; set; }
-
+        private Guid _correlationId;
+        public Guid CorrelationId
+        {
+            get
+            {
+                return _correlationId;
+            }
+            set
+            {
+                _correlationId = value;
+                RequestContext = this.CreateRequestContext(value);
```

Comment:
>RequestContext [](start = 16, length = 14)

request context should only be created when API call is made. this will cause issues in a multithreaded case